src/Sylius/Behat/Context/Domain/ManagingProductsContext.php

```diff
@@ -46,7 +45,7 @@ public function iTryToDeleteTheVariantOfProduct(ProductVariantInterface $product
     {
         try {
             $this->productVariantRepository->remove($productVariant);
-        } catch (DBALException $exception) {
+        } catch (\Exception $exception) {
```

Comment:
This is not the same, old exception still exists, it was renamed to `Doctrine\DBAL\Exception`

Reply:
Probably should be:
```suggestion
        } catch (DBALException|Exception $exception) {
```
With old & new class, or two catches...

Reply:
Due to the fact they've renamed `\Doctrine\DBAL\DBALException` to `\Doctrine\DBAL\Exception` and we consider it in the context of Behat, I decided to use generic `\Exception` as any exception thrown by `remove()` means something failed. I'm against using such generic types but in this case, I have no value added in handling a specific type (as it'll require two if's checking which one is existing). What do you think @lchrusciel?

Reply:
Luckily, we do not mute this exception entirely but store them in a shared context, so the difference is not that huge. But as we can be closer to the previous catch, don't be too generic and do it without any problem I would vote for these two catches

Reply:
@lchrusciel as in the other cases to support both Exceptions we need to add two ifs checking if a concrete exception class exists. In my opinion, it's not worth it.